src/ahbicht/expressions/condition_expression_parser.py

```diff
@@ -53,3 +57,48 @@ def parse_condition_expression_to_tree(condition_expression: str) -> Tree:
         ) from eof
 
     return parsed_tree
+
+
+def find_prerequisites(tree_or_list: Union[Tree, List[str]], sanitize: bool = False) -> ContentEvaluationPrerequisites:
+    """
+    find different types of condition nodes inside the given tree or list of keys.
+    The types are differentiated by their number range.
+    See 'Allgemeine Festlegungen' from EDI@Energy.
+    """
+    result = ContentEvaluationPrerequisites(
+        format_constraint_keys=[], requirement_constraint_keys=[], hint_keys=[], package_keys=[]
+    )
+    condition_keys: List[str]
+    if isinstance(tree_or_list, list):
+        condition_keys = tree_or_list
+    else:  # it's tree, hopefully
```

Comment:
Shouldn't we raise an error here if it's not a tree?

Reply:
I think there would be an arrow because the method `scan_values` only exists for tree like objects.

Reply:
I think there would be an arrow because the method `scan_values` only exists for tree like objects.

Reply:
so what todo? return silently?

Reply:
cdaedf06f16ea50266c69e802b8b9a5a2401627e so zufrieden @hf-krechan ?